File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> v;
        for(int i=1;i<=numRows;i++){
            vector<int> vec = {1};
            int a = 1;
            for(int j=1;j<i;j++){
                a = a * (i - j) / j;
                vec.push_back(a);
            }
            v.push_back(vec);
        }
        return v;
    }
};

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        l = []
        for i in range(1, numRows+1):
            temp = [1]
            a = 1
            for j in range(1, i):
                a = a * (i - j)//j
                temp.append(a)
            l.append(temp)
        return l